File: LiveDataDemo/app/src/main/java/guandong/com/livedatademo/repository/MyRepository.java
package guandong.com.livedatademo.repository;

import android.arch.lifecycle.MutableLiveData;

import java.util.List;

/**
 * @author guandongchen
 * @date 2018/3/22
 */
public interface MyRepository {

     MutableLiveData<List<String>> getLiveData();

     List<String> getAllData() ;

}

File: LiveDataDemo/app/src/main/java/guandong/com/livedatademo/repository/MyRepository.kt
package guandong.com.livedatademo.repository

import android.arch.lifecycle.MutableLiveData

/**
 * @author guandongchen
 * @date 2018/3/22
 */
interface MyRepository {

    val liveData: MutableLiveData<List<String>>

    val allData: List<String>

}
